store/storetest/cluster_discovery_store.go

```diff
@@ -29,13 +29,11 @@ func testClusterDiscoveryStore(t *testing.T, ss store.Store) {
 		Type:        "test_test",
 	}
 
-	if err := ss.ClusterDiscovery().Save(discovery); err != nil {
-		t.Fatal(err)
-	}
+	err := ss.ClusterDiscovery().Save(discovery)
+	require.NoError(t, err)
```

Comment:
Because we use a custom error for the application the `NoError` call is not working fine, so we normally use `Nil` and `NotNil` for asserting the errors.